### ISSUE TITLE
.NET Core Support

### ISSUE THREAD

**Issue by bchavez** [paiden/Nett#12]
Would be cool...

```
The dependency Nett 0.4.2 does not support framework .NETCoreApp,Version=v1.0.
```

**Comment by paiden** [comment]
Although it would be cool to have support for .Net core in the near future, I probably will not add it before 2017 for various reasons.

1. The tools around .Net core are not finished
  I will not fiddle around with projx formats as it is obsolete already. I will definitely wait until the  .Net core project format based on MSBuild is finalized (probably this fall with VS2016)
2. As 'Nett' relies heavily on reflection, I maybe will wait until .Net core supports the full .Net reflection API again so that I don't have to ifdef code.

**Comment by ngm** [comment]
Would it be possible to integrate this as a custom provider for the new .NET Core configuration system?

https://docs.asp.net/en/latest/fundamentals/configuration.html#writing-custom-providers

It would be nice to have the TOML config work nicely with other config sources.

**Comment by paiden** [comment]
@ngm This is something already on my backlog. But also for this feature I will wait until .Net core is fully .Net Standard 2.0 compatible (e.g. the old Reflection API is available again in .Net Core).

**Comment by tomascw** [comment]
Can verify that it works on 2.0. Just installed the latest Visual Studio 2017 preview and moved my libraries that use Nett to .Net Standard 2.0. I even tested the functionality by generating a config file using .NET Core 2.0.

**Comment by bojanrajkovic** [comment]
I actually just started on a configuration provider this weekend--I've got it mostly complete, with tests to ensure everything is getting read in the way that the ASP.NET Core configuration APIs expect.

I need to add support for customizing the conversions (for example, specifying format specifiers for integers/floats/datetimes/etc.) and fix up some of the other formatting issues (specify invariant cultures, etc.), but I should have it done this weekend. I'm happy to contribute it as a PR to Nett!
